atc/db/pipeline.go

```diff
@@ -972,6 +975,56 @@ func (p *pipeline) CreateOneOffBuild() (Build, error) {
 	return build, nil
 }
 
+func (p *pipeline) CreateStartedBuild(plan atc.Plan) (Build, error) {
+	tx, err := p.conn.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer Rollback(tx)
+
+	metadata, err := json.Marshal(plan)
+	if err != nil {
+		return nil, err
+	}
+
+	encryptedPlan, nonce, err := p.conn.EncryptionStrategy().Encrypt(metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	build := &build{conn: p.conn, lockFactory: p.lockFactory}
+	err = createBuild(tx, build, map[string]interface{}{
+		"name":         sq.Expr("nextval('one_off_name')"),
+		"pipeline_id":  p.id,
+		"team_id":      p.teamID,
+		"status":       BuildStatusStarted,
+		"start_time":   sq.Expr("now()"),
+		"schema":       "exec.v2",
+		"private_plan": encryptedPlan,
+		"public_plan":  plan.Public(),
+		"nonce":        nonce,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = build.saveEvent(tx, event.Status{
```

Comment:
Saving an event here feels a little awkward. When does this normally get sent? Will it be sent twice now?

I wonder if it would be better to create it 'pending' and have the tracker also process pending one-off builds?

Reply:
I agree, but i think this is a larger discussion around build state. Seems like we need an intermediate `planned` state.

This means the `BuildStarter` actually becomes a `BuildPlanner`.